src/main/java/com/internship/backend/config/UsernamePasswordAuthenticationProvider.java

```diff
@@ -30,7 +30,7 @@ public class UsernamePasswordAuthenticationProvider implements AuthenticationPro
     public Authentication authenticate(Authentication authentication) throws AuthenticationException {
         String username = authentication.getName();
         String password = authentication.getCredentials().toString();
-        Optional<User> optionalUser = userRepository.findByUsername(username);
+        Optional<User> optionalUser = Optional.ofNullable(userRepository.findByUsername(username));
```

Comment:
I think that we can use type inference here: https://docs.oracle.com/en/java/javase/17/language/local-variable-type-inference.html

Also, we can use static import for ofNullable(...) method